### PR TITLE
All 1461/layers

### PR DESCRIPTION
**Pull request by clintonlunn**
Combining a legend and layerlist to have more central functionality. 

Basic legend display for layers and sublayer
Support for mapimagelayer and grouplayer sublayer/layer nesting
Addition of mapimagelayer legend layer list functionality
Use of sublayers for improved visibility control
Extraction of preview generation logic into a custom hook
Refactored layercontrol functionality into components

### CODE REVIEW

**Comment by jjhi11** [comment]
@clintonlunn no preview url?

**Comment by clintonlunn** [comment]
> @clintonlunn no preview it's  there, just up in the middle of the page.

**Comment by jjhi11** [comment]
1. Why do the labels show up in the legend for the qfaults?
2. If the default visibility for a layer is set to false will it show in the legend?

**Comment by clintonlunn** [comment]
@jjhi11 Labels is visible because of a sublayer has the visible property set to true. https://github.com/UGS-GIO/geohaz-v2/pull/16/files#diff-6ce9cc7a06ea0b6ecbbb7daba963b694b005d917e42afae3a14ccbdef0880103R227-R231

It was the same in the old map https://github.com/UGS-GIO/geohaz/blob/master/map.js#L1136-L1143

If you set initial visibility to off, the visibility toggle will show as off on map load

image

image

**Comment by marshallrobinson** [comment]
Overall, this is a good looking proof of concept and a great step forward. I'm not totally sure what some of your commits addressed, so I'd like to meet for an hour maybe on Monday with you/Lance/Jay to get a walkthrough of your code.

The qFaults legend and layer don't show up for me. Also, the "Labels" layer (directly beneath qFaults) doesn't need to be shown as something that can be toggled on/off - I'm going off of memory, but I believe the current Hazards Portal has the Labels service baked into the qFaults on/off toggle.

File: src/components/SidebarComponents/Layers.tsx [hunk 1]
```diff
@@ -1,57 +1,21 @@
...
-  return (
-    // <CalciteAccordion scale='l' selection-mode='multi'>
-    //   {Array.from({ length: 5 }, (_, i) => (
-    //     <CalciteAccordionItem item-title={`Layer ${i + 1}`} key={i}>
-    //       <p>Content for Layer {i + 1}</p>
-    //     </CalciteAccordionItem>
-    //   ))}
-    // </CalciteAccordion>
+const Layers: React.FC = () => {
+  const layerList = useCustomLayerList();
```
Comment (clintonlunn) [inline src/components/SidebarComponents/Layers.tsx:7]: abstract out the layerlist creation to a custom hook at [/src/hooks/useCustomLayerList.tsx](https://github.com/UGS-GIO/geohaz-v2/pull/16/files#diff-d6c06cb6acf1c59f9aad2ed18001ea61902a54885646a33db9aa702a7becd30e)

File: src/components/SidebarComponents/Layers.tsx [hunk 2]
```diff
@@ -1,57 +1,21 @@
...
+
+  if (!layerList?.length) {
+    return <div>Loading layers...</div>;
+  }
 
-    <>
```
Comment (clintonlunn) [inline src/components/SidebarComponents/Layers.tsx:23]: removing placeholder content

File: src/components/SidebarComponents/Layers.tsx [hunk 3]
```diff
@@ -1,57 +1,21 @@
...
-      >
-        <CalciteIcon scale='s' slot='icon' icon='map-pin' />
-        <CalciteNotice open>
-          <div slot='message'>The viewers are going to love this</div>
-        </CalciteNotice>
-      </CalciteBlock>
-    </>
-  )
-}
+  return (
```
Comment (clintonlunn) [inline src/components/SidebarComponents/Layers.tsx:13]: replace with actual layerList

File: src/components/LayerControls.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { CalciteSliderCustomEvent, CalciteSwitchCustomEvent } from "@esri/calcite-components";
+
+interface LayerControlsProps {
+    layerVisibility: boolean | undefined;
+    handleVisibilityToggle: (event: CalciteSwitchCustomEvent<void>) => void | undefined;
+    layerOpacity: number;
+    handleOpacityChange: (e: CalciteSliderCustomEvent<void>) => void;
+}
+
+const LayerControls: React.FC<LayerControlsProps> = ({ layerVisibility, handleVisibilityToggle, layerOpacity, handleOpacityChange }) => (
```
Comment (clintonlunn) [inline src/components/LayerControls.tsx:11]: A controls component that can takes in some handlers that affect whichever layer we want to toggle visibility/opacity for

File: src/config/layers.ts [hunk 5]
```diff
@@ -215,6 +215,7 @@ const qFaultsConfig: LayerProps = {
     options: {
         sublayers: [
             {
+                title: 'Quaternary_Faults',
```
Comment (clintonlunn) [inline src/config/layers.ts:218]: this needs to match the name in the layer that we get back from the [fetch request](https://github.com/UGS-GIO/geohaz-v2/pull/16/files#diff-003ef9ba2137b028fa5739ee164e167f1a3f8bcbf8c735a36d3fddb2136bfa63R86-R94)

**Review by marshallrobinson — APPROVED** [review]
Great proof of concept and progress! 

We have discussed a couple of bugs (MapImageLayer legend not showing up) and possible performance enhancements that we have recorded in Jira that will be addressed in future PRs.

**Pull request merged**